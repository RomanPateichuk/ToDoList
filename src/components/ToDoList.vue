<template>
  <div class="wrapper">
    <div class='wrapper__main'>
      <header class="header"><h1>to do list</h1></header>   
    <main class="main">
        <div class="tasks__wrapper">
        <div class="task">
         <input type="checkbox" class="check__done" id='check1' />
         <label for='check1' class="task__text"> Task 1</label>
         <input type="checkbox" class="check__remove" id="task__remove1"/>
         <label for='task__remove1' class="task__remove"></label>
        </div>
        <div class="task">
           <input type="checkbox" class="check__done" id='check2'/>
           <label for='check2' class="task__text"> Task 2</label>
           <input type="checkbox" class="check__remove" id="task__remove2"/>
           <label for='task__remove2' class="task__remove"></label>
        </div>
        <div class="task"> 
          <input type="checkbox" class="check__done" id='check3'/>
          <label for='check3' class="task__text">Task 3</label>
          <input type="checkbox" class="check__remove" id="task__remove3"/>
          <label for='task__remove3' class="task__remove"></label>
        </div>
        <div class="task new__task">
          <p class="task__text">Add a new task</p>
        </div>
      </div>
    </main>
        <footer class="footer">
        <div class="footer__progress">1/3 left</div>
        <div class="footer__btn-wrapper">
          <button class="btn">All</button>
          <button class="btn">Active</button>
          <button class="btn">Completed</button>
        </div>
      </footer> 
    </div>
    <img src="../assets/img/background_img.png" alt="background">
  </div>
</template>

<style scoped lang="scss">

.wrapper{
   @media (max-width: 280px) { 
      font-size: 1rem;
  }
  max-width: 73.625rem;
  padding: 0 5% 0 5%;
  margin: 5% auto 0 auto;
  
  img{
    z-index: 1;
    position: absolute;
    right: 9%;
    top: -6%;
    
  }
 
  .wrapper__main{
    background-color: white;
    max-width: 33.75rem;
    margin: 0 auto;
    z-index: 2;
    position: relative;
   
      .header{
        font-family: 'Open Sans', sans-serif;
        background-color: #FFCA93;
        text-align: center;
        height: 3.125rem;

          h1{
          font-size: 1.25rem;
          line-height: 2.5rem;
          }
      }

      .main{
        display: flex;
        padding: 0 1.875rem 0 1.875rem;
        flex-direction: column;
        @media (max-width: 450px) { 
          padding: 0 .5rem 0 .5rem;
        } 
          .tasks__wrapper{
            margin: 1.875rem 0;
              .task{   
              display: flex;
              background: #FFDFBE;
              border-radius: .625rem;
              padding: 0 1.25rem 0 1.25rem;
              height: 3.125rem;
              justify-content: flex-start;
              align-items: center;
              position: relative;

              .check__done, .check__remove{   
                appearance: none;
                &:checked{
                  + label{
                   &:after{
                     opacity: 1;
                   }
                  }
                }
              }

              .task__text{ 
                color: #7F4B13;
                padding: 0  1.4375rem  0 3rem;
            
                &:before{   
                  content: "";
                  display: block;
                  width: 1.5rem;
                  height: 1.5rem;
                  background: #FEB567;
                  border-radius: 5px;
                  position: absolute;
                  top: .75rem;
                  left: 1.25rem;
                }
    
                &:after {
                  content: '';
                  display: block;
                  width: 9.43px;
                  height: 13.71px;
                  border: 1px solid #FFFFFF;
                  border-left: none;
                  border-top: none;
                  position: absolute;
                  transform: rotate(45deg);
                  top: .9375rem;
                  left: 1.75rem;
                  opacity: 0; 
                }
              }
               
              &:not(:first-child){
              margin: 1.5625rem 0 0 0;
              }
          
              :last-child{
              margin-left: auto;
              }
              
              .task__remove{
                &:before{   
                  content: url("../assets/img/close_disable.png");
                  border: none;
                  position: absolute;
                  left: 90%;
                  top: 25%;
                   @media (max-width: 370px) { 
                   left: 85%
                  }
                }
    
                &:after {
                  content: url("../assets/img/close_active.png");
                  border: none;
                  position: absolute;
                  left: 90%;
                  top: 25%;
                  opacity: 0;
                   @media (max-width: 370px) { 
                   left: 85%
                  }
                }
              }
              
              }  
              
              .new__task{
              background: #FFECD8; 
              border: 1.5px dashed #FFCA93;
              justify-content: center; 
                       
              .task__text{
                 color: #FFCA93; 
                 padding: 0  1.4375rem  0 1.4375rem; 

                  &:after{
                   display: none;
                 }

                  &:before{
                    display: none;
                  }
                }
              
                 :last-child{
                  margin-left: 0;
                 }

              } 
          }
        }

      .footer{
        display: flex;
        height: 3.125rem;
        justify-content: space-between;
        align-items: center;
        background: #FFCA93;
        padding: 0 1.875rem 0 1.875rem;
        font-family: 'Open Sans', sans-serif;
         @media (max-width: 410px) { 
         font-size: 1rem;
         padding: 0 1rem 0 1rem;
          }
          @media (max-width: 314px) { 
          flex-direction: column;
          }
      
          *{
            color: rgba(127, 75, 19, .42)
          }
      
          .footer__btn-wrapper{
            :not(:last-child){
              margin-right: 1.5625rem;
                 @media (max-width: 520px) { 
                  margin-right: 0;
              }
            }
        
            .btn{
              border: 1px solid rgba(0, 0, 0, 0);
              box-sizing: border-box;
              padding: .625rem;
               @media (max-width: 520px) { 
                  padding: .3125rem
              }
            }
          .btn:hover{
              border: 1px solid #C9955D;
              border-radius: 10px;
            }

          }
        }
  }
}
</style>
