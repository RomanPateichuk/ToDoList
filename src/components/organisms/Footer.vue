<template>
  <footer class="footer">
    <FooterProgress />
    <FooterBtnWrapper />
  </footer>
</template>

<script>
export default {
  components: {
    FooterBtnWrapper: () => import("@/components/atoms/FooterBtnWrapper.vue"),
    FooterProgress: () => import("@/components/molecules/FooterProgress.vue"),
  },
};
</script>

<style scoped lang="scss">
@import "@/assets/scss/main.scss";
.footer {
  display: flex;
  min-height: 3.125rem;
  justify-content: space-between;
  align-items: center;
  background: $task-remove;
  padding: 0 1.875rem;
  font-family: "Open Sans", sans-serif;
  @media (max-width: 410px) {
    font-size: 1rem;
    padding: 0 1rem;
  }
  @media (max-width: 314px) {
    flex-direction: column;
  }

  * {
    color: $footer-font-color;
  }
}
</style>
