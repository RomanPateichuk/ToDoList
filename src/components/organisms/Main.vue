<template>
  <main :class="$style.main">
    <div :class="$style.tasks">
      <Task
        v-for="(item, index) in getFilter"
        :id="item.id"
        :key="index"
        :isChecked="item.checked"
        :TaskValue="item.value"
      />
      <div v-if="get_tasks_active == 0">Новых задач нет</div>
      <AddTask />
    </div>
  </main>
</template>

<script>
import EventBus from "../../event-bus";
import { mapGetters } from "vuex";
export default {
  mounted() {
    EventBus.$on("SetFilter", (FooterBtnValue) => {
      this.SetFilter(FooterBtnValue);
    });

    EventBus.$on("CallDeleteTask", (id) => {
      this.DeleteTask(id);
    });

    EventBus.$on("CallCompleteTask", (id) => {
      this.CompleteTask(id);
    });

    if (localStorage.tasks) {
      this.$store.commit("UpdateTasks");
    }
  },
  watch: {
    tasks: function () {
      localStorage.setItem(
        "tasks",
        JSON.stringify(this.$store.getters.get_tasks_active)
      );
    },
  },
  components: {
    Task: () => import("@/components/molecules/TaskWrapper.vue"),
    AddTask: () => import("@/components/molecules/AddTaskWrapper.vue"),
  },
  computed: {
    ...mapGetters(["getFilter", "get_tasks_active"]),

    tasks() {
      return this.$store.getters.get_tasks_active;
    },
  },

  methods: {
    SetFilter: function (FooterBtnValue) {
      this.$store.commit("setFilter", FooterBtnValue);
    },

    DeleteTask: function (id) {
      this.$store.commit("DeleteTask", id);
    },

    CompleteTask: function (id) {
      this.$store.commit("CompleteTask", id);
    },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
.main {
  display: flex;
  padding: 0 1.875rem 0 1.875rem;
  flex-direction: column;
  @include MediaMainMobile;

  .tasks {
    margin: 1.875rem 0 0 0;
  }
}
</style>
