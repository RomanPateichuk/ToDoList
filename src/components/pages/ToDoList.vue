<template>
  <div :class="$style.wrapper">
    <div :class="$style.wrapper__main">
      <Header />
      <Main />
      <Footer />
    </div>
    <img
      src="@/assets/img/background_img.png"
      alt="background"
      :class="$style.bg_img"
    />
  </div>
</template>

<script>
export default {
  components: {
    Header: () => import("@/components/organisms/Header.vue"),
    Main: () => import("@/components/organisms/Main.vue"),
    Footer: () => import("@/components/organisms/Footer.vue"),
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
.wrapper {
  @media (max-width: 410px) {
    font-size: 1rem;
  }
  max-width: 73.625rem;
  padding: 0 1rem;
  margin: 2.5rem auto 0 auto;

  .bg_img {
    z-index: 1;
    position: absolute;
    right: 4rem;
    top: -2rem;
  }

  .wrapper__main {
    background-color: $main;
    max-width: 33.75rem;
    margin: 0 auto;
    z-index: 2;
    position: relative;
  }
}
</style>
