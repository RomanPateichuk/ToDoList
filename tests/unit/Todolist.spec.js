import { shallowMount, createLocalVue } from "@vue/test-utils";
import ToDoList from "../../src/components/pages/ToDoList.vue";
import Main from "../../src/components/organisms/Main.vue";
import Footer from "../../src/components/organisms/Footer.vue";
//import Header from "../../src/components/organisms/Header.vue"
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

const mutations = {
  completeTask: jest.fn(),
  deleteTask: jest.fn(),
  saveEditTask: jest.fn(),
};

const getters = {
  getCheckedValue: () => jest.fn(),
  getFilter: () => jest.fn(),
  getTasks: () => jest.fn(),
  getDoneCount: () => jest.fn(),
};

const store = new Vuex.Store({ mutations, getters });

describe("Тестирование компонента TaskWrapper:", () => {
  const wrapper = shallowMount(ToDoList, {
    store,
    localVue,
  });

  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Подключение компонента 'Main'", () => {
    expect(wrapper.findComponent(Main).exists()).toBe(true);
  });

  test("Подключение компонента 'Footer'", () => {
    expect(wrapper.findComponent(Footer).exists()).toBe(true);
  });

  test("Подключение компонента 'Header'", () => {
    //expect(wrapper.findComponent(Header).exists()).toBeTruthy()
  });
});
