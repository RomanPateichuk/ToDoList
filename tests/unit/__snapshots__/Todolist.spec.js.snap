// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента TaskWrapper: соответствие снимку 1`] = `
<div>
  <div>
    <header>
      <h1>to do list</h1>
    </header>
    <main>
      <div>
        <!---->
        <div><input type="text" placeholder="Add a new task"></div>
      </div>
    </main>
    <footer>
      <div>
        function mockConstructor() {
        return fn.apply(this, arguments);
        }/0 left
      </div>
      <div>
        <!---->
        <!---->
        <!---->
      </div>
    </footer>
  </div> <img src="@/assets/img/background_img.png" alt="background">
</div>
`;
